cmdstanpy/utils.py

```diff
@@ -957,9 +961,21 @@ def scan_sampling_iters(
 ) -> int:
     """
     Parse sampling iteration, save number of iterations to config_dict.
+    Also save number of divergences, max_treedepth hits
     """
     draws_found = 0
     num_cols = len(config_dict['column_names'])
+    idx_divergent = None
+    idx_treedepth = None
+    max_treedepth = None
+    ct_divergences = 0
+    ct_max_treedepth = 0
+    try:
+        idx_divergent = config_dict['column_names'].index('divergent__')
+        idx_treedepth = config_dict['column_names'].index('treedepth__')
+        max_treedepth = config_dict['max_depth']
+    except ValueError:
+        pass
```

Comment:
I'm curious about this logic - under what condition would this exception be thrown?

Is there any way to avoid then checking if `max_treedepth` is None for every line as we iterate through?

Reply:
this code is generalized for samples which could be generated by "fixed_param" (program has no model block) or standalone generated quantities (model block isn't executed).

we could make 2 methods - one which expects sample diagnostics and one which doesn't - would that be cleaner?

Reply:
I believe the function which calls this takes in an argument indicating that it is fixed_param - it would at least be cleaner if this function did too, even if that doesn’t fix check-per-loop issue. Splitting it into two functions wouldn’t be too bad either

Reply:
refactored.